### PR TITLE
refactor(package): Use format strings

### PR DESCRIPTION
**Pull request by jletey**
#### Description

This PR deprecates `symbol` `style` keys in `PackageConfig`, and adds `format` key instead.

Based off of #1063.

#### Motivation and Context

Related #1057

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Screenshots (if appropriate):

Screenshot 2020-04-08 at 17 27 59

#### How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

File: src/configs/package.rs [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
+use crate::config::{ModuleConfig, RootModuleConfig};
 
-use ansi_term::{Color, Style};
 use starship_module_config_derive::ModuleConfig;
 
 #[derive(Clone, ModuleConfig)]
 pub struct PackageConfig<'a> {
-    pub symbol: SegmentConfig<'a>,
-    pub style: Style,
+    pub format: &'a str,
```
Comment (heyrict) [inline src/configs/package.rs:7]: Same as #1066

File: src/modules/package.rs [hunk 2]
```diff
@@ -1,34 +1,38 @@
...
+    let config: PackageConfig = PackageConfig::try_load(module.config);
+    let formatter = if let Ok(formatter) = StringFormatter::new(config.format) {
+        formatter.map(|variable| match variable {
+            "version" => Some(module_version.clone()),
+            _ => None,
+        })
+    } else {
+        log::warn!("Error parsing format string in `package.format`");
+        return None;
+    };
```
Comment (heyrict) [inline src/modules/package.rs:28]: Ditto

**Comment by andytom** [comment]
Hi @jletey, just a friendly reminder about this PR. If you need some help just shout.

**Comment by heyrict** [comment]
I've updated your branch to the latest changes. Thank you for initial efforts :) Going to merge this now.

**Comment by jletey** [comment]
@heyrict Ah darn, really sorry for going radio silent. Couldn't thank you enough for taking the time to finish these up!

**Comment by heyrict** [comment]
@jletey Nice to see you again here :) It is a hard time for many of use and it really does not matter. Looking forward for your next contributions here!

**Pull request merged**